packages/contracts-bedrock/scripts/fpac/FPACOPS.s.sol

```diff
@@ -0,0 +1,141 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.15;
+
+import { Proxy } from "src/universal/Proxy.sol";
+import { IDisputeGame } from "src/dispute/interfaces/IDisputeGame.sol";
+import { StdAssertions } from "forge-std/StdAssertions.sol";
+import "scripts/Deploy.s.sol";
+
+/// @notice Deploys the Fault Proof Alpha Chad contracts.
+contract FPACOPS is Deploy, StdAssertions {
+    ////////////////////////////////////////////////////////////////
+    //                        ENTRYPOINTS                         //
+    ////////////////////////////////////////////////////////////////
+
+    function deployFPAC(address _proxyAdmin, address _systemOwnerSafe) public {
+        console.log("Deploying a fresh FPAC system and OptimismPortal2 implementation.");
+
+        prankDeployment("ProxyAdmin", msg.sender);
```

Comment:
Ah I see how you are using this now, very clever but I think it shows a lack of flexibility in the script. Another way to do this would be to set `CONTRACT_ADDRESSES_PATH=...` to a json that has those 2 values set to whatever `msg.sender` is but using the prank is more flexible bc it just works with whatever deployer key you use

Reply:
Good shout on adding a JSON - I had initially tried that but went w/ this for the flexibility as you mentioned. If you have any ideas on changing this I'm all ears :)

Reply:
What exactly is this doing?

Reply:
This is temporarily telling the deploy script that the proxy admin and system safe owner is the `msg.sender` of the deployment script. It's a bit hacky to get around the deploy script's design around assigning ownership of the deployed contracts.

With this, the DGF that gets deployed is temporarily owned by the `msg.sender` so that we can set the implementations for the game types, and then at the end in `transferDGFOwnershipFinal`, we assign ownership of the deployed contracts to the actual `ProxyAdmin` + `SystemSafeOwner`

Reply:
I see, so it's making sure ownership gets sent to the `msg.sender` for the duration of the script? Can you also double check that the `msg.sender` doesn't accidentally get assigned control of anything other than ownership of the DGF+proxy?

Reply:
This script shouldn't touch any ownable contracts - the only proxy that's deployed is the DGF one, and the dispute game factory is the only ownable. There are checks for the DGF's ownership transfer here: https://github.com/ethereum-optimism/optimism/pull/9495/files#diff-f8b386b335bccaa367f05001445ff88834fa7136ab2a71f5f4a8058b2d9beeebR86-R90

Other than that:
* `FaultDisputeGame` - no owner, minimal proxies w/ no owner/admin created by the DGF
* `PermissionedDisputeGame` - no owner, roles are directly set from the L2OO's challenger / proposer keys in the system config.
* `MIPS` - no owner, no proxy
* `PreimageOracle` - no owner, no proxy
* `OptimismPortal2` - impl only, proxy upgrade is done in https://github.com/ethereum-optimism/superchain-ops/pull/76